pagerduty/user.go

```
@@ -326,6 +354,21 @@ func (s *UserService) CreateContactMethod(userID string, contactMethod *ContactM
 	return v.ContactMethod, resp, nil
 }
 
+func (s *UserService) searchExistingContactMethod(userID string, contactMethod *ContactMethod) (*ContactMethod, *Response, error) {
+	lResp, _, lErr := s.ListContactMethods(userID)
+	if lErr != nil {
+		return nil, nil, fmt.Errorf("user contact method is not unique and failed to fetch existing ones: %w", lErr)
+	}
+
+	for _, contact := range lResp.ContactMethods {
+		if isSameContactMethod(contact, contactMethod) {
+			return s.GetContactMethod(userID, contact.ID)
+		}
+	}
+
+	return nil, nil, fmt.Errorf("user contact method address is used with different attributes (possibly label)")
```

why not return the original error?